tests/qam-updinstall/update_install.pm

```diff
@@ -208,7 +208,10 @@ sub run {
     foreach (@modules) {
         # substitue SLES_SAP for LTSS repo at this point is SAP ESPOS
         $_ =~ s/SAP_(\d+(-SP\d)?)/$1-LTSS/ if is_sle('15+');
-        $_ =~ s/SAP_(\d+(-SP\d)?)/SERVER_$1-LTSS/ if is_sle('=12-sp5') && !get_var('BUILD') =~ /saptune/;
+
+        # Do not update module name when the package is only from SLE-SAP(not LTSS)
+        my @channels = get_incident_channels($incident_id);
+        $_ =~ s/SAP_(\d+(-SP\d)?)/SERVER_$1-LTSS/ if is_sle('=12-sp5') && (grep { $_ =~ /SERVER_$1-LTSS/ } @channels);
```

Comment:
@a-kpappas pointed out that the `$1` in the `grep` command is not set by the regexp in the same line (`if` condition runs first), so the `grep` will never work. We need to change this.

Reply:
updated the code